src/adapters/RedisConfigAdapter.ts

```diff
@@ -0,0 +1,65 @@
+import { ConfigSpecs, IConfigAdapter } from './IConfigAdapter';
+import * as redis from 'redis';
+import { RedisClientOptions } from 'redis';
+
+export default class RedisConfigAdapter implements IConfigAdapter {
+  private client;
+
+  public constructor(hostname: string, port?: number, password?: string) {
+    const options: RedisClientOptions = {
+      socket: {
+        host: hostname,
+        port: port,
+      },
+      password: password
+    };
+    this.client = redis.createClient(options);
+  }
+
+  public async getConfigSpecs(): Promise<ConfigSpecs | null> {
+    if (!this.client.isOpen) {
+      await this.client.connect();
+    }
+    const obj = await this.client.hGetAll('config-specs');
+    if (obj === null) {
+      return null;
+    }
+
+    const result: {[key: string]: any} = {};
+    for (const key in obj) {
+      if (Object.hasOwnProperty.call(obj, key)) {
+        result[key] = JSON.parse(obj[key]);
+      }
+    }
```

Comment:
this puts the 3 keys together for the store?  gates/configs/layers?

Reply:
It returns a single object that looks like:
{gates: {"test_gate": {...}, ...}, configs: {"test_config": {...}, ...}, layers: {"layers": {...}, ...}, ...}